commit 6ed76f4efc856431cd8511ef881a494a13280653
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Wed Jul 17 14:56:14 2024 +0200

    htl: Fix __pthread_init_thread declaration and definition
    
    0e75c4a4634f ("hurd: Fix pthread_self() without libpthread") added a
    declaration for ___pthread_init_thread instead of __pthread_init_thread,
    and missed defining the external hidden symbol.

diff --git a/sysdeps/mach/hurd/htl/pt-pthread_self.c b/sysdeps/mach/hurd/htl/pt-pthread_self.c
index 08ef2e4956..d11d0ec676 100644
--- a/sysdeps/mach/hurd/htl/pt-pthread_self.c
+++ b/sysdeps/mach/hurd/htl/pt-pthread_self.c
@@ -24,6 +24,7 @@
 struct __pthread __pthread_init_thread = {
 	.thread = 1,
 };
+libc_hidden_def (__pthread_init_thread)
 
 __thread struct __pthread *___pthread_self = &__pthread_init_thread;
 libc_hidden_tls_def (___pthread_self)
diff --git a/sysdeps/mach/hurd/htl/pt-sysdep.h b/sysdeps/mach/hurd/htl/pt-sysdep.h
index ef7f509611..8e94395be7 100644
--- a/sysdeps/mach/hurd/htl/pt-sysdep.h
+++ b/sysdeps/mach/hurd/htl/pt-sysdep.h
@@ -34,8 +34,8 @@
 
 extern __thread struct __pthread *___pthread_self;
 libc_hidden_tls_proto (___pthread_self)
-extern struct __pthread ___pthread_init_thread;
-libc_hidden_proto (___pthread_init_thread)
+extern struct __pthread __pthread_init_thread;
+libc_hidden_proto (__pthread_init_thread)
 
 #ifdef DEBUG
 #define _pthread_self()                                            \
